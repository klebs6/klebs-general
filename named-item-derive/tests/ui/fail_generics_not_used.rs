// named-item-derive/tests/trybuild/fail_generics_not_used.rs

#![allow(unused)]
use named_item_derive::NamedItem;
use named_item::*;

// If we want to show that you can define a generic param
// but not actually store it, you might get some warnings or errors:
#[derive(NamedItem)]
pub struct UnusedGeneric<T> {
    name: String,
    // We do not store `T` anywhere else...
}

fn main() {}

