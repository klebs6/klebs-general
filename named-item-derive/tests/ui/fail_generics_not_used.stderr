error[E0392]: type parameter `T` is never used
  --> tests/ui/fail_generics_not_used.rs:10:26
   |
10 | pub struct UnusedGeneric<T> {
   |                          ^ unused type parameter
   |
   = help: consider removing `T`, referring to it in a field, or using a marker such as `PhantomData`
   = help: if you intended `T` to be a const parameter, use `const T: /* Type */` instead
